src/components/GameTile.vue

```diff
@@ -0,0 +1,350 @@
+<script setup>
+import { reactive, defineProps, computed } from "vue";
+import CustomButton from "@/components/CustomButton.vue";
+
+const props = defineProps({
+  tile: {
+    type: Object,
+    required: true,
+    default: () => null,
+  },
+  currency: {
+    type: String,
+    default: "USD",
+    required: true,
+  },
+});
+
+const platformIconStyle = computed(() => {
+  if (props.tile?.platform) {
+    try {
+      const iconUrl = require(`@/assets/GameTile/${props.tile.platform}-logo.png`);
+      return { "--platform-icon": `url(${iconUrl})` };
+    } catch (e) {
+      console.error(`Cannot load icon for platform: ${props.tile.platform}`);
+      return {};
+    }
+  }
+  return {};
+});
+
+function truncateDescription(text, maxLength) {
+  if (text.length <= maxLength) return text;
+  let truncated = text.slice(0, maxLength);
+  const lastSpace = truncated.lastIndexOf(" ");
+  if (lastSpace > 0) {
+    truncated = truncated.slice(0, lastSpace);
+  }
+  return truncated + "...";
+}
+
+const addToCartButton = reactive({
+  iconSource: require(`@/assets/GameTile/shopping-cart_white.png`),
+  iconAlternativeText: "shopping-cart_white.png",
+  label: "ADD TO CART",
+  disabled: false,
+});
+
+function isDiscount() {
+  const discountInPercentage = props.tile?.discountInPercentage ?? 0;
+  return discountInPercentage > 0;
+}
+
+function discountedPrice() {
+  let discountInPercentage = Math.max(
+    Math.min(props.tile?.discountInPercentage, 100),
+    1
+  );
+  return Math.max(
+    props.tile?.price - (props.tile?.price * discountInPercentage) / 100,
+    0.01
+  ).toFixed(2);
+}
+
+function handleAddToCartButtonDisabled() {
+  addToCartButton.disabled = true;
+  addToCartButton.iconSource = require(`@/assets/GameTile/shopping-cart_after.png`);
+  addToCartButton.iconAlternativeText = "shopping-cart_after.png";
+  addToCartButton.label = "ADDED";
+}
```

Comment:
The name is misleading.

```suggestion
function handleAddToCartButtonClick() {
  addToCartButton.disabled = true;
  addToCartButton.iconSource = require(`@/assets/GameTile/shopping-cart_after.png`);
  addToCartButton.iconAlternativeText = "PLEASE DESCRIBE THE IMAGE? HELLO?";
  addToCartButton.label = "ADDED";
}
```

Maybe I woud event create constants for the icons.
```javascript
// at the top of the script block
const cartIcon = {
  initial: require(`@/assets/GameTile/@/assets/GameTile/shopping-cart_white.png`),
  added: require(`@/assets/GameTile/shopping-cart_after.png`)
};
```
```suggestion
function handleAddToCartButtonClick() {
  addToCartButton.disabled = true;
  addToCartButton.iconSource = cartIcon.added;
  addToCartButton.iconAlternativeText = "PLEASE DESCRIBE THE IMAGE? HELLO?";
  addToCartButton.label = "ADDED";
}
```

What is more please name the icon properly. "shopping-cart_after"? What the heck is after? After what? Just provide the state of the icon - added, ig. 
With the new name convention could be: "icon_shopping-cart_added.svg"